src/Elasticsearch5/SearchAdapter/Aggregation/Interval.php

```diff
@@ -0,0 +1,238 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace ScandiPWA\CatalogGraphQl\Elasticsearch5\SearchAdapter\Aggregation;
+
+use Magento\Elasticsearch\SearchAdapter\ConnectionManager;
+use Magento\Elasticsearch\Model\Adapter\FieldMapperInterface;
+use Magento\Elasticsearch\Model\Config;
+use Magento\Elasticsearch\SearchAdapter\SearchIndexNameResolver;
+use Magento\CatalogSearch\Model\Indexer\Fulltext;
+use Magento\Elasticsearch\Elasticsearch5\SearchAdapter\Aggregation\Interval as CoreInterval;
+
+/**
+ * Aggregate price intervals for search query result.
+ */
+class Interval extends CoreInterval
+{
+    /**
+     * Minimal possible value
+     */
+    const DELTA = 0.005;
+
+    /**
+     * @var ConnectionManager
+     */
+    private $connectionManager;
```

Comment:
Please use the `protected` access modifiers instead of `private` in all your code.